kube-runtime/src/watcher.rs

```diff
@@ -499,6 +524,110 @@ where
             }
             let mut lp = wc.to_list_params();
             lp.continue_token = continue_token;
+            match api.list(&lp).await {
+                Ok(list) => {
+                    let last_bookmark = list.metadata.resource_version.filter(|s| !s.is_empty());
+                    let continue_token = list.metadata.continue_.filter(|s| !s.is_empty());
+                    if last_bookmark.is_none() && continue_token.is_none() {
+                        return (Some(Err(Error::NoResourceVersion)), State::Empty);
+                    }
+                    match wc.initial_list_strategy {
+                        InitialListStrategy::ListWatch => {
+                            // we have drained the last page - move on to next stage
+                            (None, State::InitPage {
+                                continue_token,
+                                objects: list.items.into_iter().collect(),
+                                last_bookmark,
+                            })
+                        }
+                        InitialListStrategy::ListWatchParallel => {
+                            // start watch
+                            match api
+                                .watch(&wc.to_watch_params(), &last_bookmark.clone().unwrap())
```

Comment:
Is this unwrap fallible?

Reply:
There's a preceding last_bookmark.is_none() check, so this unwrap is guaranteed to be safe here.

Reply:
The preceding check only fails if `continue` is also none.

Reply:
you are right. I will fix it later